test/dockertest.go

```diff
@@ -0,0 +1,77 @@
+package test
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/require"
+)
+
+func RunPostgresDocker(t *testing.T) *dockertest.Resource {
+	pool, err := dockertest.NewPool("")
+	require.NoErrorf(t, err, "failed to create dockertest pool")
+
+	resource, err := pool.RunWithOptions(
+		&dockertest.RunOptions{
+			Repository: "postgres",
+			Tag:        "13.0",
+			Env: []string{
+				"POSTGRES_USER=postgres",
+				"POSTGRES_PASSWORD=postgres",
+				"POSTGRES_DB=bubbly",
+			},
+		},
+	)
+	require.NoErrorf(t, err, "failed to start docker")
+
+	t.Cleanup(func() {
+		if err := pool.Purge(resource); err != nil {
+			t.Fatalf("Could not purge resource: %s", err)
+		}
+	})
+
+	err = waitUntilDatabaseIsReady(t, pool, resource)
+	require.NoErrorf(t, err, "error waiting for database to be ready")
+
+	return resource
+}
+
+func waitUntilDatabaseIsReady(t *testing.T, pool *dockertest.Pool, resource *dockertest.Resource) error {
+
+	pgConnStr := fmt.Sprintf("postgresql://%s:%s@localhost:%s/%s?sslmode=disable",
+		"postgres", "postgres", resource.GetPort("5432/tcp"), "bubbly")
```

Comment:
The string literals here have the same content as in the lines 19 to 23, which means we duplicate information.

I would:
✔️ Read this information from the environment variables. Or at least deduplicate it.
❓  Consider using keyword/value connection string format which is equivalent but more readable.

[Postgres 33.1.1. Connection Strings](https://www.postgresql.org/docs/13/libpq-connect.html#LIBPQ-CONNSTRING)

Reply:
Yup, because everything was hard coded with consts and I got tired of updating the Bubbly Context every time you want to do something with a DB. So I hardcoded them as literal strings to the defaults :)

Reading from env vars, or better yet Bubbly Context, would solve all this